decidim-proposals/app/models/decidim/proposals/proposal_vote.rb

```diff
@@ -4,15 +4,30 @@ module Decidim
   module Proposals
     # A proposal can include a vote per user.
     class ProposalVote < ApplicationRecord
-      belongs_to :proposal, foreign_key: "decidim_proposal_id", class_name: "Decidim::Proposals::Proposal", counter_cache: true
+      belongs_to :proposal, foreign_key: "decidim_proposal_id", class_name: "Decidim::Proposals::Proposal"
       belongs_to :author, foreign_key: "decidim_author_id", class_name: "Decidim::User"
 
       validates :proposal, uniqueness: { scope: :author }
       validate :author_and_proposal_same_organization
       validate :proposal_not_rejected
 
+      after_save :update_proposal_votes_count
+      after_destroy :update_proposal_votes_count
+
+      def self.temporary
+        where(temporary: true)
+      end
+
+      def self.final
+        where(temporary: false)
+      end
+
       private
 
+      def update_proposal_votes_count
```

Comment:
Can't we use `counter_cache` at the `belongs_to`?

Reply:
No, because it doesn't take into account the `where` condition on the other side. Counter caches are kind of broken in general as fas as I've seen.